sdk/tests/core/test_az_hfsm_pipeline.c

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_test_definitions.h"
+#include <azure/core/internal/az_hfsm_pipeline_internal.h>
+#include <azure/core/internal/az_hfsm_internal.h>
+#include <azure/core/internal/az_precondition_internal.h>
+#include <azure/core/az_result.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+
+#include <cmocka.h>
+
+static az_hfsm_pipeline az_hfsm_pipeline_test;
+
+static az_hfsm_policy P01; // Outbound
```

Comment:
Instead of adding comments, I recommend changing the names to test_policy_outbound, test_policy_middle, test_policy_inbound.
I do recommend a comment showing the layout: `//test_policy_outbound <--> test_policy_middle <--> test_policy_inbound.